src/main/java/com/ocient/jdbc/XGConnection.java

```
@@ -3976,32 +3976,42 @@ public void reconnect(final boolean shouldRequestVersion) throws IOException, SQ
 		}
 	}
 
-	private void clientHandshakeFinalize() throws SQLException, Exception {
+	// FIXME
+	// Blocking IO
+	private CompletionStage<Void> clientHandshakeFinalize() throws SQLException, Exception {
 		// Set the default schema on the connection if one has not already been set
-		resetSchema().ifPresent(schema -> {
-			try
-			{
-				setSchema(schema);
-			}
-			catch (final Exception e)
-			{
-			}
-		});
 
-		if (setPso.isPresent()) {
-			if (setPso.getAsLong() == -1)
-			{
-				// We have to turn it off
-				setPSO(false);
-			}
-			else if (setPso.getAsLong() > 0)
-			{
-				// Set non-default threshold
-				setPSO(setPso.getAsLong());
-			}
-		}
-
-		resendParameters();
+		return resetSchema()
+		    .map(schema -> {
+				try {
+					return setSchemaAsync(schema);
+				} catch (SQLException e) {
+					LOGGER.log(Level.WARNING, String.format("Exception %s occurred in clientHandshakeFinalize() with message %s", e.toString(), e.getMessage()));
```

Need to propagate the exception here